File: packages/app/src/services/software/definitions/glasswire.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'glasswire';
	public name = 'GlassWire';
	public category = [SoftwareCategory.Network, SoftwareCategory.Utility];
	public downloadName = 'GlassWireSetup.exe';
	public requiresAdmin = true;
	public icon = 'glasswire.png';
	public homepage = 'https://glasswire.com';

	public async resolveDownloadUrl() {
		return ok('https://download.glasswire.com/GlassWireSetup.exe');
	}
}

File: src/lib/software/glasswire.py
from src.lib.software import BaseSoftware

class GlassWire(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'GlassWireSetup.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.glasswire.com/GlassWireSetup.exe')
